### PR TITLE
fix: Disable role invite processing in fastboot mode

### LINKED ISSUE #3347: Error message after accepting a role-invite
I saw there was a similar bug but with different error message https://github.com/fossasia/open-event-frontend/issues/2771

**Describe the bug**
After accepting an co-organizer role invitation, the following message showed up. However, the acceptance worked. I was able to access the event page. 

Screenshot from 2019-08-02 09-32-20

**Desktop**
 - OS: Ubuntu 18.04.1
 - Browser: Chrome

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #3347

### CODE REVIEW

File: app/routes/public/role-invites.js [hunk 1]
```diff
@@ -31,6 +37,8 @@ export default Route.extend({
 
   },
   afterModel() {
+    if (this.fastboot.isFastBoot)
```
Comment (niranjan94) [inline app/routes/public/role-invites.js]: Codacy Issue found: [Expected { after 'if' condition.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=4971157)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=4971157&bid=16421692)

File: app/routes/public/role-invites.js [hunk 2]
```diff
@@ -1,8 +1,14 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
 export default Route.extend({
+  fastboot: service(),
```
Comment (niranjan94) [inline app/routes/public/role-invites.js]: Codacy Issue found: [Missing space after key 'fastboot'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=4971157)

**Comment by kushthedude** [comment]
This is more of a workaround, to ignore the fastboot crashes in the production.

**Comment by iamareebjamal** [comment]
OK,I'll be waiting for your fix

**Comment by kushthedude** [comment]
I am not saying I have fixed or this is wrong. We can atleast let the issue
open to see what is causing the crashes and fix it in near future.

**Comment by iamareebjamal** [comment]
transitionTo throws an error in fastboot mode. Not because of a bug, it is by design. It is not meant to work in fastboot mode. Only on client

Read This:
https://github.com/ember-fastboot/fastboot/issues/54

```
FastBoot should never call transitionTo
transitionTo should only be called client-side.
```

We use fastboot to render the HTML on server before it reaches client. This route only sends a POST request to API server and redirects. There is no HTML to be rendered, so why should it be handled on fastboot server?

**Comment by iamareebjamal** [comment]
Even if you replace transitionTo with replaceWith, you'll not get the route HTML - https://github.com/embermap/ember-cli-fastboot-testing/issues/204

This is the disadvantage of letting a framework do the heavy lifting for us. We just don't have enough control

**Comment by iamareebjamal** [comment]
Moar examples - https://github.com/iezer/splanky/commit/be706036e46ad5369f50b5c8e3c4662f7ad14a57

**Comment by iamareebjamal** [comment]
Issue opened since 2016 (about 4 years) - https://github.com/ember-fastboot/ember-cli-fastboot/issues/202

**Comment by iamareebjamal** [comment]
More issues - https://github.com/ember-learn/ember-api-docs/issues/128

You can try the "fix" using redirect instead of afterModel or model, but everything is a workaround when the issue is upstream - https://github.com/ember-learn/ember-api-docs/pull/129/files

**Comment by kushthedude** [comment]
Didn't know the points you mentioned about the fastboot and transitions. It
is pretty clear now the reason for the failures.👍

**Pull request merged**